File: app/src/main/java/com/example/lucas/haushaltsmanager/RecyclerView/AdditionalFunctionality/InsertStrategy/InsertStrategy.java
package com.example.lucas.haushaltsmanager.RecyclerView.AdditionalFunctionality.InsertStrategy;

import com.example.lucas.haushaltsmanager.RecyclerView.Items.IRecyclerItem;

import java.util.List;

public interface InsertStrategy {
    int INVALID_INDEX = -1;

    int insert(IRecyclerItem item, List<IRecyclerItem> items);
}

File: app/src/main/java/com/example/lucas/haushaltsmanager/RecyclerView/AdditionalFunctionality/InsertStrategy/InsertStrategy.kt
package com.example.lucas.haushaltsmanager.RecyclerView.AdditionalFunctionality.InsertStrategy

import com.example.lucas.haushaltsmanager.RecyclerView.Items.IRecyclerItem

interface InsertStrategy {
    fun insert(item: IRecyclerItem, items: MutableList<IRecyclerItem>): Int
}